packages/protocol/src/action-protocol/element-type-hints.ts

```diff
@@ -133,3 +143,70 @@ export namespace SetTypeHintsAction {
         };
     }
 }
+
+/**
+ * Response Action for a {@link RequestCheckEdgeTargetAction}. It returns
+ * a boolean indicating whether the requested element is a valid target
+ * for the edge being created.
+ */
+export interface CheckEdgeTargetResultAction extends ResponseAction {
+    kind: typeof CheckEdgeTargetResultAction.KIND;
+
+    /**
+     * true if the selected element is a valid target for this edge,
+     * false otherwise.
+     */
+    isValid: boolean;
+}
+
+export namespace CheckEdgeTargetResultAction {
+    export const KIND = 'checkEdgeTargetResult';
+}
+
+/**
+ * Send a Request to the server to check if an element is a valid target
+ * when creating a new Edge.
+ */
+export interface RequestCheckEdgeTargetAction extends RequestAction<CheckEdgeTargetResultAction> {
+    kind: typeof RequestCheckEdgeTargetAction.KIND;
+
+    /**
+     * The element type of the edge being created.
+     */
+    edgeTypeId: string;
+
+    /**
+     * The ID of the edge source element.
+     */
+    sourceElementId: string;
+
+    /**
+     * The ID of the edge target element to check.
+     */
+    targetElementId: string;
+}
+
+export namespace RequestCheckEdgeTargetAction {
+    export const KIND = 'requestCheckEdgeTarget';
+
+    export function create(
```

Comment:
For actions with more than two parameters we typically encapsulate the parameters in an `options` object:
```ts
```suggestion
     export function create(options: {
        sourceElement: SModelElement | string;
        targetElement: SModelElement | string;
        edgeTypeId: string;
    }): RequestCheckEdgeTargetAction {
        return {
            kind: KIND,
            edgeTypeId: options.edgeTypeId,
            sourceElementId: getElementId(options.sourceElement),
            targetElementId: getElementId(options.targetElement),
            requestId: ''
        };
    }
}
```